src/java.base/share/classes/java/util/EnumSet.java

```diff
@@ -178,8 +179,12 @@ public static <E extends Enum<E>> EnumSet<E> copyOf(Collection<E> c) {
             Iterator<E> i = c.iterator();
             E first = i.next();
             EnumSet<E> result = EnumSet.of(first);
-            while (i.hasNext())
-                result.add(i.next());
+            if (c instanceof RegularEnumSetCompatible || c instanceof JumboEnumSetCompatible) {
+                result.addAll(c);   // optimized for compatible sets
+            } else {
+                while (i.hasNext())
+                    result.add(i.next());
```

Comment:
Can't we just use addAll in all cases?

Reply:
I wonder why the original code didn't use `addAll`, so I didn't change it. 😂

Reply:
> I wonder why the original code didn't use `addAll`, so I didn't change it. 😂

It seems to be okay to use `addAll` in all cases. Do we need extra test cases for this?